app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,62 @@
+class RentalsController < ApplicationController
+
+  def create
+    rental = Rental.new(rental_params)
+
+    movie = Movie.find_by(id: rental_params[:movie_id])
+    customer = Customer.find_by(id: rental_params[:customer_id])
+
+    if movie.nil?
+      render json: {errors: "Movie with id #{rental_params[:movie_id]} doesn't exist"}, status: :bad_request
+      return
+    end
+
+    if movie.available_inventory == 0
+      render json: { errors: "No available inventory for #{movie.title}." }, status: :bad_request
+      return
+    end
+
+    if rental.save
+      movie.decrement_available_inventory
```

Comment:
It might be logical to move all of this updating logic into a rental model method. One thing to keep in mind in the future is that when an action requires many model updates, you may want to combine these into a *transaction*. If the rental save succeeds, then you also want to ensure that the movie and customer object updates work, otherwise you could end up in an odd state where the movie inventory is updated, but the customer's isn't.